src/renderer/components/+preferences/preferences-navigation/__tests__/preferences-navigation.test.tsx

```diff
@@ -0,0 +1,164 @@
+import { PreferencesNavigation } from "../preferences-navigation";
+import type { PreferenceNavigationItem } from "../preference-navigation-items.injectable";
+import preferenceNavigationItemsInjectable from "../preference-navigation-items.injectable";
+import { computed } from "mobx";
+import { noop } from "../../../../utils";
+import type { IComputedValue } from "mobx/dist/internal";
+
+describe("<PreferencesNavigation />", () => {
+  let di: DiContainer;
+  let render: DiRender;
+
+  beforeEach(async () => {
+    di = getDiForUnitTesting({ doGeneralOverrides: true });
+    render = renderFor(di);
+  });
+
+  it("renders", () => {
+    const { container } = render(<PreferencesNavigation />);
+
+    expect(container).toBeInTheDocument();
+  });
+
+  describe("when general navigation items passed", () => {
+    beforeEach(() => {
+      const generalNavItems: IComputedValue<PreferenceNavigationItem[]> = computed(() => [
+        {
+          id: "general",
+          label: "General",
+          isActive: computed(() => false),
+          isVisible: computed(() => true),
+          navigate: () => noop,
+          orderNumber: 0,
+        },
+        {
+          id: "proxy",
+          label: "Proxy",
+          isActive: computed(() => false),
+          isVisible: computed(() => true),
+          navigate: () => noop,
+          orderNumber: 1,
+        },
+      ]);
+
+      di.override(preferenceNavigationItemsInjectable, () => generalNavItems);
+    });
+
+    it("renders them", () => {
+      const { container } = render(
+        <PreferencesNavigation/>,
+      );
+
+      expect(container).toHaveTextContent("General");
+      expect(container).toHaveTextContent("Proxy");
+    });
+
+    it("does not show custom settings block", () => {
+      const { queryByTestId } = render(
+        <PreferencesNavigation/>,
+      );
+
+      expect(queryByTestId("extension-settings")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("when general + extension navigation items passed", () => {
+    beforeEach(() => {
+      const generalNavItems: IComputedValue<PreferenceNavigationItem[]> = computed(() => [
+        {
+          id: "general",
+          label: "General",
+          isActive: computed(() => false),
+          isVisible: computed(() => true),
+          navigate: () => noop,
+          orderNumber: 0,
+        },
+        {
+          id: "proxy",
+          label: "Proxy",
+          isActive: computed(() => false),
+          isVisible: computed(() => true),
+          navigate: () => noop,
+          orderNumber: 1,
+        },
+        // Extension navigation items
+        {
+          id: "extension-preferences-navigation-item-lensapp-node-menu",
+          label: "lensapp-node-menu",
+          isActive: computed(() => false),
+          isVisible: computed(() => true),
+          navigate: () => noop,
+          orderNumber: 0,
+          fromExtension: true,
+        },
```

Comment:
!! We should write code in a way that whether item is from Open Lens or extension shouldn't matter. Therefore stuff like `fromExtension` is not welcome to navigation items.

Reply:
We need this flag to understand where the navigation item comes from. Look at this file `src/renderer/components/+preferences/preferences-navigation/preferences-navigation.tsx` where we decided where to visually show navigation item regarding its source ("generic" or "from extension"). Is there any other way to do this without explicitly marking navigation item?

Reply:
All navigation items from extensions are renderer within separate block.

Reply:
Then, should the "block" be created with a loop as well? In that way we would have support for multiple menu "blocks" immediately. Then menu items wouldn't known anything about extensions. Menu item would know label, "block" / parent so that we can create the menu hierarchy with a loop.


Imagine you have something like this:

```
interface MenuItem {
  id: string;
  parentId: string | null;
  label: string;
  onClick: string;
}

const extensions: MenuItem = {
  id: 'extensions',
  parentId: null,
  label: 'Extensions',
  onClick: () => {}
}

const someExtension: MenuItem = {
  id: 'some-extension-id',
  parentId: 'extensions',
  label: 'Some extension',
  onClick: () => navigateToExtensionPreferences(extension)
}
```

Now you can simply loop through the items and create the UI based on that.

Reply:
Made refactoring and replaces `fromExtensions` prop to `parent`.